Add etcd installation and configuration doc

TimothyAsir opened the pull request:
tendrl-bug-id:Tendrl/node_agent#88
and #63
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing committed:
Add etcd installation and configuration doc

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsir commented:
@r0h4n, @nthomas-redhat, @nnDarshan Please review

nnDarshan commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+etcd is an open-source distributed, consistent key value store that provides a
+reliable way to store and access the data for shared configuration and service
+discovery across a cluster. Tendrl is using etcd to store the yaml files, other
+configuration files and its data.
```
We are not storing the yaml files and config files right ? 
Can we have some thing like,"tendrl uses etcd as its central store to persist all the sds information, configuration definitions and tendrl management data".

TimothyAsir replied:
Got it, I will update this

nnDarshan commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+== Installation
+etcd can be installed in any server. However for tendrl application, it can be
+installed in the server where tendrl-api is installed. The package etcd can be
+installed by running "yum install etcd"
```
should we mention from which repo is etcd available like epel for centOS and for other supported upstream platforms.

TimothyAsir replied:
Sure, i will add the availability of etcd package in epel

nnDarshan commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+the etcd service.
+Enable all the lines where there is localhost written in /etc/etcd/etcd.conf
+and replace localhost with actual IP of the machine and then start etcd server
+
```
IP or hostname should be fine right ?

TimothyAsir replied:
Done

nnDarshan commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+After updating the configuration file, the service can be started by running:
+service etcd start
+And can be verify the service by running "service etcd status"
+
```
Users can verify if the etcd is properly functioning by accessing etcd via API from browser, the api is as follows: http://<ip-addr/hostname>:2379/v2/keys/ . This should give a basic json output.

TimothyAsir replied:
Done

shtripat replied:
Or as an option user can run a curl command itself to verify..

TimothyAsirJeyasing committed:
Add etcd installation and configuration doc

1) Update etcd usage in tendrl
2) Add epel repo installation
3) Add steps to verify etcd running properly

tendrl-bug-id:Tendrl/node_agent#88 and #63
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

nnDarshan commented on etcd-configuration.adoc:
```diff
@@ -4,19 +4,22 @@
-etcd can be installed in any server. However for tendrl application, it can be
-installed in the server where tendrl-api is installed. The package etcd can be
-installed by running "yum install etcd"
+The etcd package are available in epel. The epel repo can be installed using
```
etcd package is available ?

mbukatov replied:
Good point, [I don't see any epel `etcd` builds](https://apps.fedoraproject.org/packages/etcd), but [`etcd` build for CentOS 7 is available in Extras repository](http://centos-packages.com/7/package/etcd/) directly.

TimothyAsir replied:
Yes, got it. The package is included in the CentOS Extras repository, enabled by default.

nnDarshan reviewed (commented):
One small typo. Apart from that  Looks good to me

shtripat commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+Enable all the lines where there is localhost written in /etc/etcd/etcd.conf
+and replace localhost with actual IP of the machine and then start etcd server
+
+ETCD_LISTEN_PEER_URLS="http://<ip-where-etcd-running>:2380"
```
Put this section under pre-formatted one using ```

TimothyAsir replied:
Done

shtripat commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+ETCD_ADVERTISE_CLIENT_URLS="http://<ip-where-etcd-running>:2379"
+
+After updating the configuration file, the service can be started by running:
+service etcd start
```
Same as above

TimothyAsir replied:
Done

shtripat commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+
+After updating the configuration file, the service can be started by running:
+service etcd start
+And can be verify the service by running "service etcd status"
```
%s/And can be verify the service by running/The running status of service etcd can be verified using/g

TimothyAsir replied:
Done

shtripat commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+etcd.yml is a configuration file used by tendrl-api to locate and connect
+with etcd. This file will be installed by an rpm in /etc/tendrl directory.
+Whereas this file may exist in config directory of the place where its
+deployed thru bundler installation.
```
Avoid using slangs like "thru"...

Also the above paragraph is somehow not very crisp. It could be re-phrased as below --

"The tendrl-api service stores the configuration under /etc/tendrl/etcd.yml and the same should be updated to refer correct etcd server running on a specific port."

TimothyAsir replied:
Sure, By default tendrl-api will use the configuration under config/etcd.yml from where its deployed. However for production environment it uses /etc/tendrl/etcd.yml if available. RPM deploys this file in /etc/tendrl. This configuration should be updated to refer correct etcd server running on a specific port.

@anivargi Could you share your thoughts...

shtripat commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+  :password: ''
+
+tendrl-node-agent::
+tendr-node-agent uses /etc/tendrl/tendrl.conf file to get the etcd details
```
Is this document complete? Should we have some details about tendrl.conf related to etcd configuration?

TimothyAsir replied:
Sure, We can add some more like

/etc/tendrl/tendrl.conf is used by tendrl-node-agent. Port number and ip address / hostname details should be configured in this file for all the node agents.
ex:-
etcd_port = 2379
etcd_connection = <ip address or hostname>

Could you share you suggestion here?

shtripat replied:
You can mention something like

```
The details specific to etcd are

etcd_port = 2379
etcd_connection = <IP>
```

TimothyAsir replied:
Sure

TimothyAsir commented on etcd-configuration.adoc:
```diff
@@ -0,0 +1,51 @@
+ETCD_ADVERTISE_CLIENT_URLS="http://<ip-where-etcd-running>:2379"
+
+After updating the configuration file, the service can be started by running:
+service etcd start
```
Done

TimothyAsir commented:
@shtripat, @nnDarshan, @anivargi , Please review

shtripat commented:
I dont think all the comments are taken care

TimothyAsirJeyasing committed:
Add etcd installation and configuration doc

1) Update etcd usage in tendrl
2) Add epel repo installation
3) Add steps to verify etcd running properly

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsir commented:
@r0h4n , @nthomas-redhat , @shtripat , @nnDarshan , please review
